External/Plugins/ASCompletion/Completion/ASComplete.cs

```diff
@@ -396,60 +394,101 @@ private static void AutoStartCompletion(ScintillaControl Sci, int position)
                     if (canComplete) HandleDotCompletion(Sci, true);
                 }
             }
-        }
+        }
         #endregion
-
+
         #region add_closing_braces
-        public static void HandleAddClosingBraces(ScintillaControl sci, char c, bool addedChar)
-        {
-            if (!ASContext.CommonSettings.AddClosingBraces)
-                return;
-
-            if (IsTextStyle(sci.BaseStyleAt(sci.CurrentPos - 2)) || IsInterpolationExpr(sci, sci.CurrentPos - 2))
-            {
-                foreach (Braces braces in AddClosingBracesData)
-                {
-                    if (addedChar)
-                        HandleAddBrace(sci, c, braces);
-                    else
-                        HandleRemoveBrace(sci, c, braces);
-                }
-            }
-        }
+
+        public static void HandleAddClosingBraces(ScintillaControl sci, char c, bool addedChar)
+        {
+            if (!ASContext.CommonSettings.AddClosingBraces) return;
+            
+            if (addedChar)
+            {
+                // Get the before & after style values unaffected by the entered char
+                sci.DeleteBack();
+                sci.Colourise(0, -1);
+                byte styleAfter = (byte) sci.BaseStyleAt(sci.CurrentPos);
+                byte styleBefore = (byte) sci.BaseStyleAt(sci.CurrentPos - 1);
+                sci.AddText(1, c.ToString());

```

Comment:
What's the cost of `sci.Colourise()`?
Previously (not the commit) I had the `Colourise` method to only refresh the current line, but all other usages of the method colours the whole document.
![image](https://cloud.githubusercontent.com/assets/13545633/14147002/c296e612-f6d4-11e5-93a6-133930a07946.png)
I haven't tested if there is any problem occurring when styling a line only, but if it doesn't cost too much to just colourise the whole document (seeing how `ASComplete.OnChar()` calls it every time), I'd just leave it as is.